torchao/quantization/quantize_/workflows/float8/float8_tensor.py

```diff
@@ -0,0 +1,609 @@
+
+        return addmm_float8_unwrapped_inference(
+            inpt_data,
+            input_scale,
+            w_data,
+            w_scale,
+            output_dtype=input_tensor.dtype,
+            bias=bias,
+            use_fast_accum=scaled_mm_config.use_fast_accum,
+        ).reshape(out_shape)
+    else:
+        return torch.nn.functional.linear(
+            input_tensor, weight_tensor.dequantize(), bias
+        )
+
+
+@implements(torch.bmm)
+def _(func, types, args, kwargs):
+    input_tensor, weight_tensor = (
+        args[0],
+        args[1],
+    )
+    if not isinstance(weight_tensor, Float8Tensor):
+        raise NotImplementedError("Only support handling Float8 weight")
+
+    orig_act_size = input_tensor.size()
+    act_quant_kwargs = weight_tensor.act_quant_kwargs
+    if act_quant_kwargs is not None:
+        input_tensor = _choose_quant_func_and_quantize_tensor(
+            input_tensor, act_quant_kwargs
+        )
+
+    if isinstance(input_tensor, Float8Tensor):
+        a_data = input_tensor._data
+        a_scale = input_tensor.scale
+
+        b_data = weight_tensor._data
+        b_scale = weight_tensor.scale.squeeze(-1)
+        assert b_data.is_contiguous(), "weight for bmm must be contiguous"
+
+        assert (
+            all(x == 1 for x in weight_tensor.block_size[:-1])
+            and weight_tensor.block_size[-1] == weight_tensor.shape[-1]
+        ), "bmm only works for per row weight quantization"
+        assert (
+            all(x == 1 for x in input_tensor.block_size[:-1])
+            and input_tensor.block_size[-1] == input_tensor.shape[-1]
+        ), "bmm only works for per row activation quantization"
+
+        orig_out_features = b_data.shape[-2]
+
+        res = torch.ops.fbgemm.f8f8bf16_rowwise_batched(
+            a_data,
+            b_data,
+            a_scale,
+            b_scale,
+        )
+        res = res.reshape(*orig_act_size[:-1], orig_out_features)
+    else:
+        raise NotImplementedError(
+            "bmm only support float8 dynamic activation + float8 weight"
+        )
+
+    return res
+
+
+@implements([aten.detach.default, aten.alias.default])
+def _(func, types, args, kwargs):
+    return return_and_correct_aliasing(
+        func, args, kwargs, args[0]._apply_fn_to_data(torch.detach)
+    )
+
+
+@implements(aten.clone.default)
+def _(func, types, args, kwargs):
+    return return_and_correct_aliasing(
+        func, args, kwargs, args[0]._apply_fn_to_data(torch.clone)
+    )
+
+
+def _same_metadata(self: "Float8Tensor", src: "Float8Tensor") -> bool:
+    return (
+        isinstance(self, Float8Tensor)
+        and isinstance(src, Float8Tensor)
+        and self.shape == src.shape
+        and self._data.shape == src._data.shape
+        and self.scale.shape == src.scale.shape
+        and self.act_quant_kwargs == src.act_quant_kwargs
+        and self.dtype == src.dtype
+    )
+
+
+@implements(aten.copy_.default)
+def _(func, types, args, kwargs):
```

Comment:
thoughts about having this live on the base class?

Reply:
yeah, I think it makes sense, can do this in a separate PR

Reply:
added in https://github.com/pytorch/ao/pull/2598